docs/upgrades/v0.14-to-v1-upgrade/examples/src/test/kotlin/examples/EvaluationChanges.kt

```diff
@@ -0,0 +1,225 @@
+            else -> throw RuntimeException("Unsupported type: $type")
+        }
+        return value
+    }
+
+    @Test
+    fun `orderly materialization of semi-structured data`() {
+        // In PLK 0.14.9, semi-structured data must be materialized in a predictable order
+        // This is critical for stateful processes that depend on consistent traversal patterns
+        // which means for semi-structured type, we need to fully materialize its first element
+        // before the second one (and so on).
+        val bagValue = ExprValue.newBag(
+            listOf(
+                ExprValue.newInt(42),
+                ExprValue.newStruct(
+                    listOf(
+                        ExprValue.newString("value").namedValue(ExprValue.newString("nested"))
+                    ),
+                    org.partiql.lang.eval.StructOrdering.UNORDERED
+                )
+            )
+        )
+        val structValue = ExprValue.newStruct(
+            listOf(
+                ExprValue.newInt(42).namedValue(ExprValue.newString("id")),
+                ExprValue.newList(listOf(ExprValue.newString("a"), ExprValue.newString("b")))
+                    .namedValue(ExprValue.newString("items"))
+            ),
+            org.partiql.lang.eval.StructOrdering.UNORDERED
+        )
+        // Materialize in order for semi-structured data
+        materializeInOrder(bagValue)
+        materializeInOrder(structValue)
+        assertEquals(2, bagValue.count())
+        assertEquals(2, structValue.count())
+    }
+
+    @Test
+    fun `working with evaluation environment`() {
+        // In PLK 0.14.9, evaluation context is managed through EvaluationSession
+        // Create sessions with parameters
+        val session1 = EvaluationSession.build {
+            parameters(listOf(ExprValue.newString("a"), ExprValue.newInt(1)))
+        }
+        val session2 = EvaluationSession.build {
+            parameters(listOf(ExprValue.newString("b"), ExprValue.newInt(2), ExprValue.newBoolean(true)))
+        }
+        // Combine sessions -- create a new session with combined parameters
+        val combinedParams = session1.parameters + session2.parameters
+        val combinedSession = EvaluationSession.build {
+            parameters(combinedParams)
+        }
+        assertEquals(true, combinedSession.parameters[4].scalar.booleanValue())
+        // Access parameters by index
+        val param1 = session1.parameters[0]
+        val param2 = session2.parameters[1]
+        assertEquals("a", param1.scalar.stringValue())
+        assertEquals(2L, param2.scalar.numberValue())
+    }
+
+    @Test
+    fun `working with compiler pipeline builder`() {
+        // In PLK 0.14.9, compiler pipeline customization is done through custom functions and procedures
+        val pipeline = CompilerPipeline.builder()
+            .addFunction(object : ExprFunction {
+                override val signature = FunctionSignature(
```

Comment:
Would be helpful to add short comment explaining what this function does (i.e. it just returns back the string `"custom strategy"`). Similarly for the `Datum` v1 example